sycl/doc/extensions/SPIRV/SPV_INTEL_float_controls2.asciidoc

```diff
@@ -0,0 +1,265 @@
+SPV_INTEL_float_controls2
+=========================
+
+Name Strings
+------------
+
+SPV_INTEL_float_controls2
+
+Contact
+-------
+
+To report problems or to provide input on this extension, please open a new issue at:
+https://github.com/intel/llvm/issues
+
+Contributors
+------------
+
+- Gang Chen, Intel
+- Mariusz Merecki, Intel
+- Aleksander Us, Intel
+- Konstantin Vladimirov, Intel
+- Rudenko Nikita, Intel
+
+
+
+Notice
+------
+
+Copyright (c) 2020 Intel Corporation. All rights reserved.
+
+Status
+------
+
+Working Draft
+
+This is a preview extension specification, intended to provide early access to a feature for review and community feedback. When the feature matures, this specification may be released as a formal extension.
+
+Because the interfaces defined by this specification are not final and are subject to change they are not intended to be used by shipping software products. If you are interested in using this feature in your software product, please let us know!
+
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2020-12-03
+| Revision           | 4
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.5, Revision 3, Unified
+
+This extension requires SPIR-V 1.0.
+
+Overview
+--------
+
+This extension adds new execution modes and decorations to control floating-point computations.
+This extension adds execution modes with round to positive infinity and round to negative infinity default rounding
+and execution modes specifying compliance with floating-point arithmetic (IEEE 754) standard.
+This extension adds new decorations that can be applied to a function to control floating-point computations inside the function.
+
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following
+*OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_float_controls2"
+----
+
+Modifications to the SPIR-V Specification, Version 1.5, Revision 3, Unified
+---------------------------------------------------------------------------
+
+Modify Section 2.16.1, Universal Validation Rules:
+
+Modify the Entry Point rules:
+
+Add *RoundingModeRTPINTEL* and *RoundingModeRTNINTEL* to the list of execution modes in the third rule.
```

Comment:
Minor: I'd consider describing what "the third rule" is here, since the universal validation rules appear to have been changed in the latest SPIR-V 1.6 spec and I'm having a hard time finding the older SPIR-V 1.5 revision 3 spec.

Reply:
Good point, I will quote the rule:

Add *RoundingModeRTPINTEL* and *RoundingModeRTNINTEL* to the list of execution modes in the following validation rule:

- Each [OpEntryPoint](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpEntryPoint) must not set more than one of the DenormFlushToZero or DenormPreserve [execution modes](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Execution_Mode) for any given Target Width.